interfaces/apparmor/spec.go

```diff
@@ -146,6 +166,70 @@ func (spec *Specification) AddSnippet(snippet string) {
 	}
 }
 
+// RegisterSnippetKey adds a key to the list of valid keys
+func (spec *Specification) RegisterSnippetKey(key interfaces.SnippetKey) {
+	if spec.snippetKeys == nil {
+		spec.snippetKeys = make(map[string]bool)
+	}
+	if _, ok := spec.snippetKeys[key.GetValue()]; ok {
+		logger.Panicf("priority key %s is already registered", key)
+	}
+	spec.snippetKeys[key.GetValue()] = true
+}
+
+// AddPrioritizedSnippet adds a new apparmor snippet to all applications and hooks using the interface,
+// but identified with a key and a priority. If no other snippet exists with that key, the snippet is
+// added like with AddSnippet, but if there is already another snippet with that key, the priority of
+// both will be taken into account to decide whether the new snippet replaces the old one, is appended
+// to it, or is just ignored. The key must have been previously registered using RegisterSnippetKey().
+func (spec *Specification) AddPrioritizedSnippet(snippet string, key interfaces.SnippetKey, priority uint) {
+	if _, ok := spec.snippetKeys[key.GetValue()]; !ok {
```

Comment:
Please use the whole key as the map key.

Reply:
Mmm... but what happens if the interface.SnippetKey used to add the snippet is created in a different place than where it is registered? I mean: does this test always pass?

    key1 := interfaces.NewSnippetKey("key")
    key2 := interfaces.NewSnippetKey("key")
    Assert(key1 == key2, Equals, true)

Reply:
It seems to work, but just in case I added an extra test.